contracts/minter-suite/Minters/MinterPolyptych/MinterPolyptychV0.sol

```diff
@@ -0,0 +1,1066 @@
+        (uint256 _invocations, , , , , ) = genArtCoreContract.projectStateData(
+            _projectId
+        );
+
+        requireCurrentPanelIsMintedOnce(
+            _projectId,
+            _ownedNFTAddress,
+            _ownedNFTTokenId
+        );
+
+        address _artist = genArtCoreContract.projectIdToArtistAddress(
+            _projectId
+        );
+
+        uint256 _newTokenId = (_projectId * ONE_MILLION) + _invocations;
+
+        // EFFECTS
+
+        // we need to store the new token ID before it is minted so the randomizer can query it
+        IGenArt721CoreContractV3WithRandomizer _ownedNFTCoreContract = IGenArt721CoreContractV3WithRandomizer(
+                _ownedNFTAddress
+            );
+        bytes12 targetHashSeed = _ownedNFTCoreContract.tokenIdToHashSeed(
+            _ownedNFTTokenId
+        );
+        require(targetHashSeed != 0, "Cannot have an empty hash seed to copy.");
+
+        genArtCoreContract.randomizerContract().setPolyptychHashSeed(
+            _newTokenId,
+            targetHashSeed
+        );
+
+        // once mint() is called, the polyptych randomizer will either:
+        // 1) assign a random token hash
+        // 2) if configured, obtain the token hash from the `polyptychSeedHashes` mapping
+        tokenId = minterFilter.mint(_to, _projectId, vault);
+
+        // NOTE: delegate-vault handling **ends here**.
+
+        // redundant check against reentrancy
+        bytes12 _assignedHashSeed = genArtCoreContract.tokenIdToHashSeed(
+            tokenId
+        );
+        require(
+            _assignedHashSeed == targetHashSeed,
+            "Unexpected token hash seed"
+        );
+
+        // okay if this underflows because if statement will always eval false.
+        // this is only for gas optimization (core enforces maxInvocations).
+        unchecked {
+            if (tokenId % ONE_MILLION == _projectConfig.maxInvocations - 1) {
+                _projectConfig.maxHasBeenInvoked = true;
+            }
+        }
+
+        // INTERACTIONS
+        // require sender to own NFT used to redeem
+        /**
+         * @dev Considered an interaction because calling ownerOf on an NFT
+         * contract. Plan is to only register AB/PBAB NFTs on the minter, but
+         * in case other NFTs are registered, better to check here. Also,
+         * function is non-reentrant, so being extra cautious.
+         */
+        if (msg.sender == _artist) {
+            require(
+                IERC721(_ownedNFTAddress).ownerOf(_ownedNFTTokenId) == _to,
+                "Only owner of NFT"
+            );
+        } else {
+            require(
+                IERC721(_ownedNFTAddress).ownerOf(_ownedNFTTokenId) == vault,
+                "Only owner of NFT"
+            );
+        }
+
+        // split funds
+        uint256 _pricePerTokenInWei = _projectConfig.pricePerTokenInWei;
+        address _currencyAddress = _projectConfig.currencyAddress;
```

Comment:
```suggestion
        address _currencyAddress = _projectConfig.currencyAddress;
        if (_currencyAddress == address(0) {
            // ETH
            if (msg.sender == _artist) {
                require(
                    IERC721(_ownedNFTAddress).ownerOf(_ownedNFTTokenId) == _to,
                    "Only owner of NFT"
                );
            } else {
                require(
                    IERC721(_ownedNFTAddress).ownerOf(_ownedNFTTokenId) == vault,
                    "Only owner of NFT"
                );
            }
        } else {
            // ERC20
            address _owner = IERC721(_ownedNFTAddress).ownerOf(_ownedNFTTokenId);
            require(_owner == _to || owner == vault);
        }
        // split funds
        uint256 _pricePerTokenInWei = _projectConfig.pricePerTokenInWei;
```

Reply:
++; I think this is necessary to allow for the flow that we discussed w/ BMG